CodeEdit/Features/WindowCommands/FileCommands.swift

```diff
@@ -6,15 +6,59 @@
 //
 
 import SwiftUI
+import CryptoKit
 
 struct FileCommands: Commands {
+
+    @State var windowController: CodeEditWindowController?
+
+    /// Default instance of the `FileManager`
+    private let filemanager = FileManager.default
+
     var body: some Commands {
         CommandGroup(replacing: .newItem) {
             Group {
                 Button("New") {
-                    NSDocumentController.shared.newDocument(nil)
+                    guard let workspace = windowController?.workspace else {
+                        return
+                    }
+
+                    do {
+                        try filemanager.createDirectory(at: workspace.supportDirURL, withIntermediateDirectories: true)
```

Comment:
This maybe looks like a great use of macOS's temporary directory:

https://developer.apple.com/documentation/foundation/filemanager/1642996-temporarydirectory

Instead of one in application support.